exchange/src/exchange/traders/trader_types.hpp

```diff
@@ -112,52 +134,63 @@ struct remote_trader_t : public generic_trader_t {
 
 struct local_trader_t : public generic_trader_t {
     explicit local_trader_t(std::string algo_path) :
-        generic_trader_t(generate_user_id()), ALGO_PATH(std::move(algo_path))
+        generic_trader_t(std::move(algo_path))
     {}
 
-    // FOR TESTING PURPOSES ONLY
-    explicit local_trader_t(std::string user_id, std::string algo_path) :
-        generic_trader_t(std::move(user_id)), ALGO_PATH(std::move(algo_path))
-    {}
+    constexpr TraderType
+    get_type() const override
+    {
+        return TraderType::LOCAL;
+    }
 
     std::string
-    get_algo_path() const
+    get_algo_id() const override
     {
-        return ALGO_PATH;
+        return get_id();
     }
 
     void
-    set_pid(pid_t pid)
+    set_pid(pid_t pid) override
     {
         pid_ = pid;
     }
 
     pid_t
-    get_pid() const
+    get_pid() const override
     {
         return pid_;
     }
 
 private:
-    const std::string ALGO_PATH;
     pid_t pid_{};
+};
 
-    static uint
-    get_and_increment_user_id()
+struct bot_trader_t : public generic_trader_t {
+    bot_trader_t() : generic_trader_t(generate_user_id()) {}
+
+    constexpr TraderType
+    get_type() const override
     {
-        static uint user_id = 0;
-        return user_id++;
+        return TraderType::BOT;
     }
 
-    static std::string
-    generate_user_id()
+    pid_t
+    get_pid() const override
     {
-        return "LOCAL_" + std::to_string(get_and_increment_user_id());
+        return -1;
     }
-};
 
-struct bot_trader_t : public generic_trader_t {
-    bot_trader_t() : generic_trader_t(generate_user_id()) {}
+    void
+    set_pid(pid_t) override
+    {
+        throw std::runtime_error("Not implemented");
+    }
+
+    std::string
+    get_algo_id() const override
+    {
+        throw std::runtime_error("Not implemented");
```

Comment:
bots should have an algo id for debugging

just implement something stupid

Reply:
they already have a generated bot id, they don't need an algo id too